src/plugins/intel_gna/src/transformations/gather_sinking_transpose.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+#include "transformations_visibility.hpp"
+
+namespace ov {
+namespace intel_gna {
+namespace pass {
+
+/**
+ * @brief Merge Gather with Transpose through Reshape.
+ *
+ *      Any1[a,b,c]          Any1[a,b,c]
+ *       |                     |
+ *    Reshape[1,a*b*c]      Reshape[1,a*b*c]
+ *       |                     |
+ *     Gather[1,a*b*c]      Gather[1,a*b*c]
+ *       |                     |
+ *    Reshape[a,b,c]           |
```

Comment:
I think we should indicate here we're detecting patterns with possibly multiple reshapes at this position.

Reply:
updated commnets